### PR TITLE
Bug fix/m9 useless assignment to variable

### PR DESCRIPTION
**Pull request by naylanasir**
**What's in the PR**
* we have worked on the removal of useless assignments to local variables, this will increase maintainability, and improve performance by removing useless assignment operations

**How to test manually**
* ...

**Automatic testing**
* [ ] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

File: inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/recommender/NamedEntityLinker.java [hunk 1]
```diff
@@ -203,7 +203,7 @@ private void predictToken(String aCoveredText, int aBegin, int aEnd, CAS aCas)
         }
 
         Type predictedType = getPredictedType(aCas);
-        Feature scoreFeature = getScoreFeature(aCas);
+        //Feature scoreFeature = getScoreFeature(aCas);
```
Comment (munterkalmsteiner) [inline inception-concept-linking/src/main/java/de/tudarmstadt/ukp/inception/conceptlinking/recommender/NamedEntityLinker.java]: This change is not related to the suggested improvement.

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_BACKUP_133.js [hunk 2]
```diff
@@ -0,0 +1,3283 @@
+
```
Comment (munterkalmsteiner) [inline inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_BACKUP_133.js]: What are these backup files? They should not be submitted to the repository...

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js [hunk 3]
```diff
@@ -462,7 +462,7 @@
     events = [];
     for (sel in eventsObj) {
       functionName = eventsObj[sel];
-      _ref1 = sel.split(' '), selector = 2 <= _ref1.length ? __slice.call(_ref1, 0, _k = _ref1.length - 1) : (_k = 0, []), event = _ref1[_k++];
```
Comment (munterkalmsteiner) [inline inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js]: This is a generated file (see comments at the beginning of file), hence refactoring should be done in the code that generates this file. However, since this is 3rd party code, I don't think this is possible.

File: inception-imls-dl4j/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/dl4j/pos/DL4JSequenceRecommender.java [hunk 4]
```diff
@@ -381,7 +381,7 @@ public void predict(RecommenderContext aContext, CAS aCas) throws Recommendation
             Type predictedType = getPredictedType(aCas);
             Type tokenType = getType(aCas, Token.class);
 
-            Feature scoreFeature = getScoreFeature(aCas);
+            //Feature scoreFeature = getScoreFeature(aCas);
```
Comment (munterkalmsteiner) [inline inception-imls-dl4j/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/dl4j/pos/DL4JSequenceRecommender.java]: Better to delete the code, instead of just commenting it out...

File: inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/QualifierFeatureEditor.java [hunk 5]
```diff
@@ -377,7 +377,7 @@ private void setSelectedKBItem(KBHandle value, Item<LinkWithRoleModel> aItem,
         if (linkedAnnotationFeature == null) {
             linkedAnnotationFeature = getLinkedAnnotationFeature();
         }
-        List<KBHandle> handles = new ArrayList<>();
+        List<KBHandle> handles;
```
Comment (munterkalmsteiner) [inline inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/QualifierFeatureEditor.java]: This is likely not correct. The method will now potentially return null, which might not be expected.

File: inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/QualifierFeatureEditor.java [hunk 6]
```diff
@@ -467,7 +467,7 @@ private void actionAdd(AjaxRequestTarget aTarget)
             List<LinkWithRoleModel> links = (List<LinkWithRoleModel>) QualifierFeatureEditor.this
                 .getModelObject().value;
             AnnotatorState state = QualifierFeatureEditor.this.stateModel.getObject();
-
+			state=null;
```
Comment (munterkalmsteiner) [inline inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/QualifierFeatureEditor.java]: Why setting state to null here?

File: inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/SubjectObjectFeatureEditor.java [hunk 7]
```diff
@@ -359,7 +359,7 @@ private KBHandle getSelectedKBItem()
         if (linkedAnnotationFeature == null) {
             linkedAnnotationFeature = getLinkedAnnotationFeature();
         }
-        List<KBHandle> handles = new ArrayList<>();
```
Comment (munterkalmsteiner) [inline inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/feature/SubjectObjectFeatureEditor.java:362]: Same comment as earlier.

File: inception-scheduling/src/main/java/de/tudarmstadt/ukp/inception/scheduling/config/SchedulingProperties.java [hunk 8]
```diff
@@ -44,6 +44,6 @@ public int getQueueSize()
 
     public void setQueueSize(int aQueueSize)
     {
-        queueSize = queueSize;
+        queueSize = aQueueSize;
```
Comment (munterkalmsteiner) [inline inception-scheduling/src/main/java/de/tudarmstadt/ukp/inception/scheduling/config/SchedulingProperties.java]: Hm, this might be actually a genuine bug that you fixed! However, it is not related to your refactoring...

File: inception-recommendation-api/src/test/java/de/tudarmstadt/ukp/inception/recommendation/api/recommender/IncrementalSplitterTest.java [hunk 9]
```diff
@@ -117,6 +117,7 @@ public void thatSplittingWorks() {
         @Test(expected = IllegalArgumentException.class)
         public void thatPercentageHasToBePercentage() {
             IncrementalSplitter splitter = new IncrementalSplitter(42.1337, 1, 10);
+			splitter=null;
```
Comment (munterkalmsteiner) [inline inception-recommendation-api/src/test/java/de/tudarmstadt/ukp/inception/recommendation/api/recommender/IncrementalSplitterTest.java]: If the variable is not used, better delete it instead of setting to null...

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]
There are a couple of issues with this PR, please see comments.

**Comment by naylanasir** [comment]
Hi, 
We have addressed all the changes you suggested in the review…
These changes are pushed for your kind consideration.

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js (hunk 3 above)
Comment (munterkalmsteiner) [reply inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js]: There should be no changes in this file.

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_BASE_133.js [hunk 10]
```diff
@@ -0,0 +1,3274 @@
+
```
Comment (munterkalmsteiner) [inline inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_BASE_133.js]: Please remove this file.

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_LOCAL_133.js [hunk 11]
```diff
@@ -0,0 +1,3277 @@
+
```
Comment (munterkalmsteiner) [inline inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_LOCAL_133.js]: Please remove this file.

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_REMOTE_133.js [hunk 12]
```diff
@@ -0,0 +1,3275 @@
+
```
Comment (munterkalmsteiner) [inline inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_REMOTE_133.js]: Please remove this file.

File: inception-imls-lapps/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/lapps/LappsRecommenderIntegrationTest.java [hunk 13]
```diff
@@ -120,7 +120,7 @@ private Dispatcher buildDispatcher()
             public MockResponse dispatch(RecordedRequest request) {
                 try {
                     String url = request.getPath();
-                    String body = request.getBody().readUtf8();
+                    //String body = request.getBody().readUtf8();
```
Comment (munterkalmsteiner) [inline inception-imls-lapps/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/lapps/LappsRecommenderIntegrationTest.java]: Don't just comment out code, delete the line if it is not necessary. Do this with all.

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]
Ok, 3 comments more. Almost there.

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js (hunk 3 above)
Comment (naylanasir) [reply inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full.js]: Changes in this file are undone

**Comment by naylanasir** [comment]
1)All the changes in annotator.js have been undone
2) changes to worker.js have also been undone, as it is also a third party code, liscenced to be used as is..
2)The three unwanted files, i.e. annotator-full_REMOTE133.js, annotator-full_BASE133.js and annotator-full_LOCAL133.js are removed.
3) commented out useless assignments have been deleted

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_BASE_133.js (hunk 10 above)
Comment (naylanasir) [reply inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_BASE_133.js]: removed

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_LOCAL_133.js (hunk 11 above)
Comment (naylanasir) [reply inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_LOCAL_133.js]: file has been removed

File: inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_REMOTE_133.js (hunk 12 above)
Comment (naylanasir) [reply inception-html-editor/src/main/java/de/tudarmstadt/ukp/inception/htmleditor/annotatorjs/resources/annotator-full_REMOTE_133.js]: file removed

File: inception-imls-lapps/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/lapps/LappsRecommenderIntegrationTest.java (hunk 13 above)
Comment (naylanasir) [reply inception-imls-lapps/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/lapps/LappsRecommenderIntegrationTest.java]: useless assignment deleted

Comment (naylanasir) [reply inception-imls-lapps/src/test/java/de/tudarmstadt/ukp/inception/recommendation/imls/lapps/LappsRecommenderIntegrationTest.java]: deleted this commented out useless assignment

**Comment by naylanasir** [comment]
All the suggested changes have been addressed....
The chart.js file has been edited to match the file in this repository,
but it now has conflicts with the upstream repository.

File: inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/chart/resources/chart.js [hunk 14]
```diff
@@ -22,58 +22,58 @@
...
+      // make the type of x-axis "category" (shows x in category intervals of size 1). 
     // It is for better visualization when the x-axis represents test data size (annotationpage)
     if ((arrayOfLearningCurves[0][2] - arrayOfLearningCurves[0][1]) > 1) 
-    	xAxixType = "category";
+      xAxixType = "category";
     
 
     // if we just have one value per data-row, we cannot visualize a step
     var plotTypes = {};
-	for (i = 0; i < arrayOfLearningCurves.length; i++) {
```
Comment (munterkalmsteiner) [inline inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/chart/resources/chart.js:42]: These changes here are not so nice as they are not needed... just add noise to the review.

**Review by munterkalmsteiner — APPROVED** [review]
Thanks! Most of the changes makes sense. The changes in `chart.js` are a bit difficult to follow. No explanation on what has been changed why (changes seem not to be trivial).
